Logger.md

```diff
@@ -0,0 +1,119 @@
+# Logger
+
+`Logger` is an interface that helps decouple client classes from concrete implementations of a logging system.
+
+## Declaration
+
+```swift
+public enum LogLevel {
+    case info
+    case warning
+    case error
+}
+
+public protocol Logger {
+  func log(_ level: LogLevel, tag: String, message: String)
+}
+```
+
+```kotlin
+// Kotlin
+TODO
+```
+
+## Usage
+
+```swift
+// Swift
+
+#if DEBUG
+  let logger = DeviceConsoleLogger()
+#else
+  let logger = BugfenderLogger() // This class is an example 
+#endif
+
+let manager = NetworkManager(logger) // This class is an example 
+
+```
+
+## Implementation
+
+```swift
+// Swift
+// Swift
+/// Information that gets logged can be labeled as one of the following informative labels.
+public enum LogLevel {
+    case info
+    case warning
+    case error
+}
+
+/// Abstracts concrete implementations of a logger system.
+public protocol Logger {
+    
+    /// Logs a String object using a given level
+    ///
+    /// - Parameters:
+    ///   - level: Type of log.
+    ///   - tag: An additional label to help categorise logs.
+    ///   - message: The message to be logged.
+    func log(_ level: LogLevel, tag: String?, message: String)
+}
+
+// MARK: - Default implementations
+public extension Logger {
+    
+    func info(tag: String? = nil, message: String) {
```

Comment:
--> `_ message: String`

Reply:
Ok, changed. In general I am in favour of using method labels, though.

Reply:
I do agree, but for such trivial and explicit parameter as the message to log, in a logger class, in the log method, it is just too redundant. And also, "message" it is character consuming (Too long).

`MyLogger.log(message: "myMessage")` vs `MyLogger.log("MyMessage")`